package-res/resources/web/prompting/PromptPanel.js

```diff
@@ -354,6 +354,10 @@ define(['cdf/lib/Base', 'cdf/Logger', 'dojo/number', 'dojo/i18n', 'common-ui/uti
               return paramValue.toUpperCase() != paramSelectedValue.toUpperCase();
             case "java.sql.Date": // Set time to zero to eliminate its influence on the days comparison
               return (new Date(paramValue).setHours(0,0,0,0)) != (new Date(paramSelectedValue).setHours(0,0,0,0));
+            case "[Ljava.lang.String;": // Comparing String arrays
```

Comment:
This only handles "Steelwheels" cases (ie you have no real data model and all column keys are strings). I'd rather see a generic array test instead. If paraType starts with "[" we know we have an array. The members of the array will be strings (thanks to our XML parser), so the comparison is still valid as it is. But now we also handle integer-arrays and all other common types used in database keys.